src/api/operator/numpy/np_elemwise_broadcast_logic_op.cc

```diff
@@ -44,31 +44,65 @@ MXNET_REGISTER_API("_npi.not_equal")
   UFuncHelper(args, ret, op, op_scalar, nullptr);
 });
 
+void set_UFuncHelper(runtime::MXNetArgs args, runtime::MXNetRetValue* ret,
+                     const nnvm::Op* op, const nnvm::Op* op_scalar,
+                     const nnvm::Op* op_rscalar) {
+  int result = 0;
+  if (args[1].type_code() == kNDArrayHandle) {
+    result++;
+    result <<= 1;
+  }
+  if (args[0].type_code() == kNDArrayHandle) {
+    result++;
+  }
+
+  switch (result) {
```

Comment:
this is a clever trick, yet I suggest to explicitly use `if (..&&..) ... else if (.. && ..) ... else` to make it more clear and readable.